### PR TITLE
new feature 

### PR DESCRIPTION
**Pull request by berslen**
boolean updateSelectedHistoryItemDateTime: selected history item to top of the list.
callback for onHistoryItemSelected

It moves selected history item to the top of the list.
After selecting history item onHistoryItemSelected called

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -31,6 +32,16 @@ class MyApp extends StatelessWidget {
                 decoration: InputDecoration(hintText: 'List type'),
               ),
 
+              /// sample1
```
Comment (trashfeed) [inline example/lib/main.dart]: "sample 1” is duplicated. Please use a different name.
I suggest moving it to the bottom as “sample5.” 🙏

File: example/lib/main.dart [hunk 2]
```diff
@@ -31,6 +32,16 @@ class MyApp extends StatelessWidget {
...
+              /// - list with updateHistoryItemDateTime
+              InputHistoryTextField(
+                historyKey: "05",
+                listStyle: ListStyle.List,
+                onHistoryItemSelected: (value) => print(value),
+                updateSelectedHistoryItemDateTime: true,
+                decoration: InputDecoration(hintText: 'List type'),
```
Comment (trashfeed) [inline example/lib/main.dart]: `List type` is duplicated. please different `hintText`.

```suggestion
                decoration: InputDecoration(hintText: 'List type(update in descending order)'),
```

File: example/lib/main.dart [hunk 3]
```diff
@@ -103,7 +114,7 @@ class MyApp extends StatelessWidget {
                 listTextStyle: TextStyle(fontSize: 30),
                 historyListItemLayoutBuilder: (controller, value, index) {
                   return InkWell(
-                    onTap: () => controller.select(value.text),
+                    onTap: () async => await controller.select(value.text),
```
Comment (trashfeed) [inline example/lib/main.dart]: older versions don't have asyc-await, maybe don't work ?

File: pubspec.yaml [hunk 4]
```diff
@@ -12,6 +12,7 @@ dependencies:
   flutter:
     sdk: flutter
   shared_preferences: ^2.0.8
+  rxdart: ^0.28.0
```
Comment (trashfeed) [inline pubspec.yaml]: sorry, but we do not want to introduce new libraries. Is it possible to achieve this with the current library?

**Review by trashfeed — CHANGES_REQUESTED** [review]
Thanks for the PR. I've made a couple of comments that I'd like you to check out.

File: example/lib/main.dart (hunk 3 above)
Comment (berslen) [reply example/lib/main.dart:117]: _save() added to select method which needs to be await

File: pubspec.yaml (hunk 4 above)
Comment (berslen) [reply pubspec.yaml:15]: i could only resolve the sink issue with this library

**Comment by berslen** [comment]
> Thanks for the PR. I've made a couple of comments that I'd like you to check out.

Made necessary fixes

**Review by trashfeed — APPROVED** [review]
@berslen 
Thanks for the PR! it a great feature.

**Pull request merged**